packages/netlify-cms-widget-markdown/src/serializers/remarkShortcodes.js

```diff
@@ -8,19 +8,64 @@ export function remarkParseShortcodes({ plugins }) {
   methods.unshift('shortcode');
 }
 
+export function getLinesWithOffsets(value) {
+  const SEPARATOR = '\n\n';
+  const splitted = value.split(SEPARATOR);
+  const trimmedLines = splitted
+    .reduce(
+      (acc, line) => {
+        const { start: previousLineStart, originalLength: previousLineOriginalLength } =
+          acc[acc.length - 1];
+
+        return [
+          ...acc,
+          {
+            line: line.trimEnd(),
+            start: previousLineStart + previousLineOriginalLength + SEPARATOR.length,
+            originalLength: line.length,
+          },
+        ];
+      },
+      [{ start: -SEPARATOR.length, originalLength: 0 }],
+    )
+    .slice(1)
+    .map(({ line, start }) => ({ line, start }));
+  return trimmedLines;
+}
+
 function createShortcodeTokenizer({ plugins }) {
   return function tokenizeShortcode(eat, value, silent) {
-    let match;
-    const potentialMatchValue = value.split('\n\n')[0].trimEnd();
-    const plugin = plugins.find(plugin => {
-      match = value.match(plugin.pattern);
-
-      if (!match) {
-        match = potentialMatchValue.match(plugin.pattern);
-      }
+    // Plugin patterns may rely on `^` and `$` tokens, even if they don't
+    // use the multiline flag. To support this, we fall back to searching
+    // through each line individually, trimming trailing whitespace and
+    // newlines, if we don't initially match on a pattern. We keep track of
+    // the starting position of each line so that we can sort correctly
+    // across the full multiline matches.
+    const trimmedLines = getLinesWithOffsets(value);
 
-      return !!match;
-    });
+    // Attempt to find a regex match for each plugin's pattern, and then
+    // select the first by its occurrence in `value`. This ensures we won't
+    // skip a plugin that occurs later in the plugin registry, but earlier
+    // in the `value`.
+    const { plugin, match } =
+      plugins
+        .toList()
+        .map(plugin => ({
+          match:
+            value.match(plugin.pattern) ||
+            trimmedLines
+              .map(({ line, start }) => {
+                const match = line.match(plugin.pattern);
+                if (match) {
+                  match.index += start;
+                }
+                return match;
+              })
+              .find(match => !!match),
```

Comment:
Would it make sense to break early from the `map` when we find a match? Since we're using `find` we only look for the first match correct?

Reply:
Done in https://github.com/netlify/netlify-cms/pull/5597/commits/a0605cfc312b2473dbccb8887468aeea73ae01da